chain/network/src/store/mod.rs

```diff
@@ -144,6 +144,19 @@ impl Store {
     pub fn list_peer_states(&self) -> Result<Vec<(PeerId, KnownPeerState)>, Error> {
         self.0.iter::<schema::Peers>().collect::<Result<_, _>>().map_err(Error)
     }
+
+    pub fn set_recent_connections(
+        &mut self,
+        recent_connections: &Vec<PeerId>,
+    ) -> Result<(), Error> {
+        let mut update = self.0.new_update();
+        update.set::<schema::RecentConnections>(&(), &recent_connections);
```

Comment:
this semantics doesn't match the design doc: "last 40 connections which lasted for at least 10min", since it prunes immediately any lost connections, even if they were reliable for a very long time. If you want to introduce a different semantics, please update/extend the design doc and provide a reasoning why it is sufficient and/or better.

Reply:
nvm, I can see that you implement this logic elsewhere.